qiskit/transpiler/passmanager.py

```diff
@@ -417,6 +419,13 @@ def _validate_stages(self, stages: Iterable[str]) -> None:
                 for invalid_stage in invalid_stages[1:]:
                     msg.write(f", {invalid_stage}")
                 raise ValueError(msg.getvalue())
+        if len(tuple(stages)) != len(set(stages)):
+            repeated_stages = {s for s in stages if stages.count(s) > 1}
+            warn(
+                f"Stage names {repeated_stages} provided several times, only first occurances preserved.",
+                UserWarning,
```

Comment:
We should document the behavior about duplicated stages in the docstring and that it will warn when this occurs.

Reply:
I don't seem to find any other file were `UserWarnings` show up in the docstring. What would be the preferred format for this?

Reply:
Yeah, most of qiskit doesn't emit a lot of warnings besides deprecation warnings (which should hopefully be clearly documented).

I think the important part here isn't that a user warning is being emitted though, but more that we document how the function behaves with duplicated stage names. So having a section in the docstring that says, "if duplicated stage names are used this function will do X". You can also mention a warning will be emitted as part of that if you want.